File: chapter6/php/com/dotComIt/learnWith/tests/AuthenticationTest.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: jhouser
 * Date: 12/6/2017
 * Time: 4:15 PM
 */

/*require_once '../config/DatabaseConfig.php';
require_once '../services/AuthenticationService.php';*/

require_once dirname(__FILE__) . '/../config/DatabaseConfig.php';
require_once dirname(__FILE__) . '/../services/AuthenticationService.php';



$databaseConfig = new DatabaseConfig();
$conn = $databaseConfig->openConnection();

$authenticationService = new AuthenticationService($conn);

echo('<h1>Test Authentication Success</h1>');
$result = $authenticationService->authenticate('me','AB86A1E1EF70DFF97959067B723C5C24');
//$result = $authenticationService->authenticate('me',hash('md5', 'me'));
echo(json_encode($result) );
echo('<br/>');
echo('<br/>');
echo('<h1>Test Authentication Fail</h1>');
$result = $authenticationService->authenticate('me','me');
echo(json_encode($result) );
echo('<br/>');
echo('<br/>');
?>

File: chapter6/java/src/main/java/com/dotComIt/learnWith/tests/AuthenticationTest.java
package com.dotComIt.learnWith.tests;


import com.dotComIt.learnWith.config.DatabaseConfig;
import com.dotComIt.learnWith.services.AuthenticationService;
import com.dotComIt.learnWith.vos.ResultObjectVO;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("test/login")
public class AuthenticationTest {

    private DatabaseConfig dc;
    private AuthenticationService as;

    public AuthenticationTest(){
        System.out.println("In Authentication Test Constructor");
        dc = new DatabaseConfig();
        as = new AuthenticationService(dc.openConnection());
    }

    // get method for success
    @GET
    @Path("success")
    @Produces(MediaType.APPLICATION_JSON)
    public ResultObjectVO loadUserSuccess() {
        return as.authenticate("me","AB86A1E1EF70DFF97959067B723C5C24");
    }


    // get method for failure
    @GET
    @Path("fail")
    @Produces(MediaType.APPLICATION_JSON)
    public ResultObjectVO loadUserFail() {
        // non hashed password so should fail
        return as.authenticate("me","me");
    }
}
